sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py

```diff
@@ -113,27 +117,24 @@ def send_events(self, events, **kwargs):
         if not isinstance(events, list):
             events = cast(ListEventType, [events])
 
-        if all(isinstance(e, CloudEvent) for e in events) or all(_is_cloud_event(e) for e in events):
+        if isinstance(events[0], CloudEvent) or _is_cloud_event(events[0]):
```

Comment:
are we only checking that the first event is a cloud event, rather than all the events, because the service will catch it if they are not all cloud events being sent?

Reply:
this is intentional - service doesn't have partial success - so either all are cloud events are not - and we avoid more client side validation.